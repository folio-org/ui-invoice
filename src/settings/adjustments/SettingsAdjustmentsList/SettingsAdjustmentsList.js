import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import {
  Button,
  NavList,
  NavListItem,
  Pane,
} from '@folio/stripes/components';

class SettingsAdjustmentsList extends Component {
  static propTypes = {
    label: PropTypes.object.isRequired,
    adjustments: PropTypes.arrayOf(PropTypes.object),
    rootPath: PropTypes.string.isRequired,
  };

  static defaultProps = {
    adjustments: [],
  };

  render() {
    const { label, rootPath, adjustments } = this.props;

    const lastMenu = (
      <Button
        to={`${rootPath}/create`}
        buttonStyle="primary paneHeaderNewButton"
        marginBottom0
      >
        <FormattedMessage id="ui-invoice.button.new" />
      </Button>
    );

    return (
      <Pane
        id="setting-adjustments-pane"
        lastMenu={lastMenu}
        paneTitle={label}
        defaultWidth="fill"
        fluidContentWidth
      >
        <NavList>
          {adjustments.map(d => (
            <NavListItem
              key={d.id}
              to={`${rootPath}/${d.id}/view`}
            >
              {d.title}
            </NavListItem>
          ))}
        </NavList>
      </Pane>
    );
  }
}

export default SettingsAdjustmentsList;
