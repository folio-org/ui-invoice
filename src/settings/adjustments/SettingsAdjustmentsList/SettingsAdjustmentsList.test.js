import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import { adjustment } from '../../../../test/jest/fixtures';

import SettingsAdjustmentsList from './SettingsAdjustmentsList';

const defaultProps = {
  label: <span>adjustments</span>,
  adjustments: [adjustment],
  rootPath: '/path',
};
const renderSettingsAdjustmentsList = (props = defaultProps) => render(
  <SettingsAdjustmentsList {...props} />,
  { wrapper: MemoryRouter },
);

describe('SettingsAdjustmentsList', () => {
  beforeEach(() => {
    global.document.createRange = global.document.originalCreateRange;
  });

  afterEach(() => {
    global.document.createRange = global.document.mockCreateRange;
  });

  it('should render correct structure', async () => {
    const { asFragment } = renderSettingsAdjustmentsList();

    expect(asFragment()).toMatchSnapshot();
  });
});
