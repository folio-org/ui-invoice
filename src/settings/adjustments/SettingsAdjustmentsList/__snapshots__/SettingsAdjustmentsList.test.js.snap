// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsAdjustmentsList should render correct structure 1`] = `
<DocumentFragment>
  <div>
    TitleManager
  </div>
  <div>
    <section
      class="pane focusIndicator"
      id="setting-adjustments-pane"
    >
      <div
        aria-labelledby="paneHeadersetting-adjustments-pane-pane-title"
        class="paneHeader"
        data-test-pane-header="true"
        id="paneHeadersetting-adjustments-pane"
        role="region"
        tabindex="-1"
      >
        <div
          class="paneHeaderCenter"
        >
          <div
            class="paneHeaderCenterInner"
            id="paneHeadersetting-adjustments-pane-pane-title"
          >
            <h2
              class="paneTitle"
              data-test-pane-header-title="true"
            >
              <span
                class="paneTitleLabel"
              >
                <span>
                  adjustments
                </span>
              </span>
            </h2>
          </div>
        </div>
        <div
          class="paneHeaderButtonsArea last"
        >
          <a
            class="button primary paneHeaderNewButton marginBottom0"
            data-test-new-adjustment-button="true"
            href="/path/create"
            role="button"
          >
            ui-invoice.button.new
          </a>
        </div>
      </div>
      <div
        class="paneContent hasPadding"
        id="setting-adjustments-pane-content"
        tabindex="-1"
      >
        <nav
          class="navListSectionControl"
          data-test-nav-list="true"
        >
          <a
            class="NavListItem"
            data-test-nav-list-item="true"
            href="/path/adjustmentId/view"
          />
        </nav>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
